python/src/nanoarrow/_lib.pyx

```diff
@@ -967,10 +989,50 @@ cdef class CSchemaBuilder:
         if self._ptr.release == NULL:
             ArrowSchemaInit(self._ptr)
 
+    @staticmethod
+    def copy_existing(CSchema existing_schema):
+        return CSchemaBuilder(existing_schema.__deepcopy__())
+
     @staticmethod
     def allocate():
         return CSchemaBuilder(CSchema.allocate())
 
+    def clear_metadata(self):
```

Comment:
Optional: What if we made this more object oriented? e.g. `CSchemaBuilder.clear_metadata()` vs `CSchemaBuilder.metadata.clear()`. Same with `append_metadata`. Does it make sense to implement this way? If not (or its a lot of work), ignore this comment.